### PR TITLE
Optimize the error message when logging in

### PR DESCRIPTION
**Pull request by rainzm**
**这个 PR 实现什么功能/修复什么问题**:
- feat(keystone): more specific error return during authentication
- feat(apigateway): differentiate error messages when logging in

- [x] 功能、bugfix描述
- [x] 冒烟测试

**是否需要 backport 到之前的 release 分支**:
- release/3.7
- release/3.6
- release/3.5

/area keystone apigateway
/cc @swordqiu @yousong

### CODE REVIEW

File: pkg/keystone/tokens/auth.go [hunk 1]
```diff
@@ -124,6 +115,15 @@ func authUserByIdentity(ctx context.Context, ident mcclient.SAuthenticationIdent
 		}
 		idpId = mapping.IdpId
 	} else {
+		// checn enable
```
Comment (ioito) [inline pkg/keystone/tokens/auth.go:118]:
```suggestion
		// check enable
```

File: pkg/keystone/tokens/auth.go [hunk 2]
```diff
@@ -99,20 +99,11 @@ func authUserByIdentity(ctx context.Context, ident mcclient.SAuthenticationIdent
...
-		if usrExt.IsLocal && usrExt.LocalFailedAuthCount > o.Options.PasswordErrorLockCount {
-			// user locked
-			return nil, errors.Wrap(httperrors.ErrUserLocked, "please contact the administrator")
-		}
-		// user disabled
-		return nil, errors.Wrap(httperrors.ErrUserLocked, "please contact the administrator")
-	}
 
-	if err == sql.ErrNoRows {
+	if err == httperrors.ErrUserNotFound {
```
Comment (ioito) [inline pkg/keystone/tokens/auth.go:106]:
```suggestion
	if err != nil && errors.Cause(err) == httperrors.ErrUserNotFound {
```

Comment (rainzm) [reply pkg/keystone/tokens/auth.go:106]: fixed,结合上文,看上去 err != nil 就足够了

File: pkg/keystone/tokens/auth.go (hunk 1 above)
Comment (rainzm) [reply pkg/keystone/tokens/auth.go:118]: fixed

**Pull request merged**